content/en/docs/How To/create_filter.md

```diff
@@ -0,0 +1,208 @@
+}
+```
+
+The [filtering example](https://github.com/AdRoll/baker/blob/main/examples/filtering/filter.go)
+shows a more complex constructor that also uses the `FilterParams` argument.
+
+### The filter configuration and help
+
+A filter requiring some configurations also has a config object, including as many keys as it
+needs and tagging each one with an `help` tag, a string that contains what a user needs to know
+which values set for it:
+
+```go
+type ClauseFilterConfig struct {
+	Clause string `help:"Boolean formula describing which events to let through. If empty, let everything through."`
+}
+```
+
+## Modify record fields
+
+A filter can change the value of the record fields before calling `next()`:
+
+```go
+func (f *NopFilter) Process(r Record, next func(Record)) {
+    var src FieldIndex = 10
+    var dst FieldIndex = 10
+    v := r.Get(src)
+    //.. modify v as required
+    r.Set(dst, v)
+    next(r)
+}
+```
+
+## Discard records
+
+The filterchain, or the ordered list of filters applied to a Record before it reaches the
+output, works thanks to the `next()` function.  
+This function is what each filter uses to decide whether a record must continue its journey in the
+filter chain (calling `next()`) or if the record should be discarded (simply not calling `next()`).
+
+When a filter discards a record it should also report it in the stats:
+
+```go
+type NopFilter struct{
+    numProcessedLines int64
+	numFilteredLines  int64
+}
+
+func (f *NopFilter) Process(r Record, next func(Record)) {
+    atomic.AddInt64(&f.numProcessedLines, 1)
+    
+    // shouldBeDiscarded is part of the filter logic
+    if shouldBeDiscarded(r) {
+        atomic.AddInt64(&f.numFilteredLines, 1)
+        // return here so next() isn't called
+        return
+    }
+    // pass the record to the next filter in the filter chain
+    next(r)
+}
+
+func (f *NopFilter) Stats() FilterStats { 
+    return baker.FilterStats{
+		NumProcessedLines: atomic.LoadInt64(&f.numProcessedLines),
+		NumFilteredLines:  atomic.LoadInt64(&f.numFilteredLines),
+    }
+}
+```
+
+## Create records
+
+A filter can call `next()` multiple times sending new or copied records to multiple filter chains.
+
+Note that the new or copied records don't start the filter chain from the first filter in the list
+but only the remaining filters are applied to the records.
+
+{{% alert color="warning" %}}
+Remember not to pass the same record to multiple `next()` functions or later changes to one of
+the records could also impact the others.  
+Always use `Copy()` or `CreateRecord()` before calling `next()` more than once.
+{{% /alert %}}
+
+### Copy()
+
+The record entering the `Process()` function can be copied calling `record.Copy()`.
+
+```go
+func (f *NopFilter) Process(r Record, next func(Record)) {
+    // Call next the 1st time
+    next(r)
+
+    // WRONG, it is the same record as above
+    next(r)
+
+    // CORRECT, this is a copy of the record
+    next(r.Copy())
+}
+```
```

Comment:
It's a good idea to illustrate the case of duplicated records but I'd be more explicit about what we're illustrating here as it's not super obvious. Maybe add something like:

> Filters can duplicate incoming records, and thus have more records come out than records that came in.